daemon/protocol/python/scoot/daemon_cli.py

```diff
@@ -0,0 +1,156 @@
+  daemon_cli.py exec poll <runIds> [--wait=<wait>] [--all]
+  daemon_cli.py exec abort <runId>
+  daemon_cli.py echo <ping>
+
+  
+Submit commands to the Scoot daemon:
+
+snapshot commands:
+create   Creates a snapshot containing the files in the srcDir directory.  
+         This command returns the id for the snapshot.
+         
+checkout Copies the contents of snapshotId to destDir.  Files from snapshotId will overwrite files in destDir.
+
+
+execution commands:
+run             Copies the files from snapshotId (if a snapshotId is supplied), and runs the command.  
+                if timeout is supplied, the daemon will wait timeout seconds for the command to complete.  
+                Default timeout is 0.5 seconds.
+
+poll            Get the status of runs.  RunIds must be a comma delimited list of run ids (1,3,5,6).
+                wait < 0 : poll will return the status values as soon as one or more run ids status is finished.
+                wait = 0 : poll will return the status values immediately
+                wait > 0 : poll will wait up to <wait> seconds for one of the run ids to reach finished state
+                If all is present, poll will return the status of all run ids, not just finished runs.
+
+abort           Abort the specified run.
+
+check daemon running:
+echo            The Scoot daemon echo's <ping> back to the client.  Use this command to verify that the daemon
+                is running.
+
+Options:
+  -h --help          Show this screen.
+  --snapshotId=<sid>  Snapshot to install before running the command.
+  --timeout=<ns>     Maximum time(ms) to allow the command to run. Default 500ms
+  --wait=<wait>      <0:wait indefinitely for at least one run to complete. 
+                     0: return immediately with the status(s).  
+                     >0: wait up to this time(seconds) for at least one of the runs to finish.  
+                     Default: 0
+  --all              Return the status of all the runs not just finished runs.
+
+"""
+import docopt
+import os
+import sys
+import re
+from scoot.docopt import DocoptExit
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../protocol'))
+import client_lib as proto
+
+def display_statuses(statuses):
+    for status in statuses:
+        print("\n\{RunId:{0},\n State:{1},\n ExitCode:{2},\n {Error:{3},\n Snapshot:{5}\n\}".format(status.runId, proto.ScootStatus.State(status.state), status.ExitCode, status.Error, status.Snapshot))
+
+
+if __name__ == '__main__':
+    try:
+        cmd = docopt.docopt(__doc__)
+    except Exception as e:
+        print("exception:{0}".format(e))
+        raise docopt.DocoptExit(e)
+    try:
+#         print("starting client")
+        proto.start()
+    except proto.ScootException as e:
+        if re.search("UNAVAILABLE", e) > 0:
+            raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+        raise docopt.DocoptExit(e)          
+        
+    if cmd['snapshot'] and cmd['create']:
```

Comment:
I'd start decomposing into functions, so that you can do:
 if cmd['snapshot']:
   snapshotCmd(cmd)
  elif cmd['exec']:
    execCmd(cmd)

Might make the structure easier to follow.